src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -595,39 +594,335 @@ public override bool IsTupleType
+                                        members.Add(new TupleRenamedElementFieldSymbol(this, fieldSymbol, defaultName, -members.Count - 1, null));
+                                    }
+
+                                    // Add a field with the given name
+                                    var location = _elementLocations.IsDefault ? null : _elementLocations[tupleFieldIndex];
+
+                                    if (field.Name == namesOfVirtualFields[tupleFieldIndex])
+                                    {
+                                        members.Add(new TupleElementFieldSymbol(this, fieldSymbol, tupleFieldIndex, location));
+                                    }
+                                    else
+                                    {
+                                        members.Add(new TupleRenamedElementFieldSymbol(this, fieldSymbol, namesOfVirtualFields[tupleFieldIndex],
+                                                                                                tupleFieldIndex, location));
+                                    }
+                                }
+                                else if (field.Name == namesOfVirtualFields[tupleFieldIndex])
+                                {
+                                    members.Add(new TupleElementFieldSymbol(this, fieldSymbol, tupleFieldIndex,
+                                                                                    _elementLocations.IsDefault ? null : _elementLocations[tupleFieldIndex]));
+                                }
+                                else
+                                {
+                                    // Add a field with default name
+                                    members.Add(new TupleFieldSymbol(this, fieldSymbol, -members.Count - 1));
+
+                                    // Add a field with the given name
+                                    members.Add(new TupleRenamedElementFieldSymbol(this, fieldSymbol, namesOfVirtualFields[tupleFieldIndex], tupleFieldIndex,
+                                                                                            _elementLocations.IsDefault ? null : _elementLocations[tupleFieldIndex]));
+                                }
+
+                                namesOfVirtualFields[tupleFieldIndex] = null; // mark as handled
+                            }
+                            else if (currentNestingLevel == 0)
+                            {
+                                // field at the top level didn't match a tuple backing field, simply add.
+                                members.Add(new TupleFieldSymbol(this, field.AsMember(currentUnderlying), -members.Count - 1));
+                            }
+                            break;
+
+                        case SymbolKind.NamedType:
+                            // We are dropping nested types, if any. Pending real need.
```

Comment:
Are types not supposed to be "inherited" like properties/events or this is just NYI?